### ISSUE TITLE
extension not working

### ISSUE THREAD

**Issue by sfabriece** [Microsoft/vscode-eslint#365]
I'm trying to setup a simple nodejs project but the extension is throwing errors.

config: 

```
'use strict';

module.exports = {
    'extends': 'eslint-config-hapi',
    rules: {
        'no-undef': 'error'
    }
};
```

error:

```
[Info  - 15:51:44] ESLint server is running.
[Info  - 15:51:49] ESLint library loaded from: {path_to_project}\node_modules\eslint\lib\api.js
[Error - 15:51:50] 
Failed to load plugin react: Cannot find module 'eslint-plugin-react'
Happened while validating {path_to_project}\index.js
This can happen for a couple of reasons:
1. The plugin name is spelled incorrectly in an ESLint configuration file (e.g. .eslintrc).
2. If ESLint is installed globally, then make sure 'eslint-plugin-react' is installed globally as well.
3. If ESLint is installed locally, then 'eslint-plugin-react' isn't installed correctly.
```

package.json: 

```
  "devDependencies": {
    "eslint": "4.13.1",
    "eslint-config-hapi": "11.1.0",
    "eslint-plugin-hapi": "4.1.0"
  },
```

**Comment by dbaeumer** [comment]
@sfabriece what happens when you run eslint from a teminal. If this works without problem can you proivide a Github repository that I can clone which demos the problem with steps on how to reproduce.

**Comment by sfabriece** [comment]
it works fine from the terminal. Let me try to put together a repo.

**Comment by sfabriece** [comment]
@dbaeumer here is a repo to reproduce: https://github.com/sfabriece/vscode-eslint-test

**Comment by dbaeumer** [comment]
This works for me without problems using the above github repository

![capture](https://user-images.githubusercontent.com/1931590/34109452-e8fe23ce-e403-11e7-8d98-12bfd5b50306.png)

Any additional info?

**Comment by sfabriece** [comment]
any special settings for eslint?

I'm using npm v5
i'm on windows 10

It might also be another plugin messing with my settings.

**Comment by dbaeumer** [comment]
I am using npm 5.5.1 and node 8.9.1. I am on Windows 10 also. 

Are you sure that in your setup eslint and all eslint plugins are either installed locally or globally. This has to be that way and is a limitation of ESLint npm module itself.

**Comment by sfabriece** [comment]
They are installed locally. It might be another extension messing with eslint. I'll try disabling all of them and see if I can identify which.

**Comment by sfabriece** [comment]
I installed vscode insider edition and only installed the eslint extension. I'm still getting the same result. I have no idea what is wrong.

**Comment by dbaeumer** [comment]
Any special paths (e.g. paths with spaces, special unicode charactes, ....)

**Comment by sfabriece** [comment]
I'll have a look and let you know.

**Comment by Lajnold** [comment]
This might happen if you have a global ~/.eslintrc that uses eslint-plugin-react. I have a problem where Visual Studio or ReSharper or something sometimes creates ~/.eslintrc upon startup, and it includes some plugin that isn't present in the project's package.json. After that, vscode-eslint doesn't work until I've removed ~/.eslintrc.

I've seen bug reports about ~/.eslintrc being used despite there being a project-local config file, and @dbaeumer hasn't been able to reproduce the problem. But it actually seems like the global config file is parsed but then ignored, except that failure in the parsing makes it break down.

**Comment by Lajnold** [comment]
@dbaeumer: Can you reproduce the problem where an invalid ~/.eslintrc stops the extension from working?

**Comment by PTBKoo** [comment]
I have the same problem but I have my eslint and all eslint plugins installed globally. It looks like vscode-eslint plugin is looking for the file locally when i have eslint globally installed. Not sure when this problem started. I also dont have a local .eslintrc file in my project.

**Comment by dbaeumer** [comment]
@Lajnold only if there is no project local config. But this is nothing VS Code can fix because in such a setup eslint on the command line doesn't succeed either.

**Comment by Lajnold** [comment]
In my case, there is a project local config. eslint on the command line works, but vscode-eslint fails.

**Comment by dbaeumer** [comment]
@Lajnold what kind of error are you seeing? This works for me with the following broken .eslintrc.json file in my home directory.

```
////
{
```

**Comment by dbaeumer** [comment]
Setup problem. I was not able to reproduce this. I see now:

![capture](https://user-images.githubusercontent.com/1931590/36373991-24ad7fa4-156a-11e8-99a8-6467ed93f565.png)

**Comment by dbaeumer** [comment]
But I don't known why this is happening since I am not instructing eslint to read the config.

**Comment by dbaeumer** [comment]
This actually fails in ESLInt itself

![capture](https://user-images.githubusercontent.com/1931590/36374379-67e0a674-156b-11e8-83ec-a20b8881c38f.png)

I don't know why in a library setup they read that file and fail and in a command line setup they don't

**Comment by dbaeumer** [comment]
The problematic part in ESLint lib is this

```
            if (localConfigDirectory === PERSONAL_CONFIG_DIR && localConfigFile !== projectConfigPath) {
```

Need to understand why this is not working when used as a library.

**Comment by dbaeumer** [comment]
OK. The problem is a path comparing issue: when using the ESLint API they tripple equal paths which in this setup are `c:\Users\dirkb` and `C:\Users\dirkb` which are obviously not equal and then they try to read the config from the user dir although it should be ignored.

**Comment by Lajnold** [comment]
Nice, thanks!

**Comment by Lajnold** [comment]
By the way, it looks like you meant to publish 1.4.7, but actually published 1.4.6. The publishing seems to be one step behind.
